In [1]:
#importing libraries
import numpy as np
import pandas as pd

pd.set_option('mode.chained_assignment', None) #to get rid of a warning

In [2]:
#specifying columns to use
usecols1 = ['CustomerID', 'CustomerName', 'CustomerCategoryID', 'BuyingGroupID', 'PrimaryContactPersonID']
usecols2 = ['CustomerCategoryID', 'CustomerCategoryName']
usecols3 = ['PersonID', 'FullName']
usecols4 = ['CustomerID', 'TransactionDate', 'TransactionAmount', 'IsFinalized']
usecols5 = ['SupplierID', 'SupplierName', 'SupplierCategoryID', 'PrimaryContactPersonID']
usecols6 = ['SupplierCategoryID', 'SupplierCategoryName']
usecols7 = ['SupplierID', 'PurchaseOrderID', 'TransactionAmount', 'IsFinalized']
usecols8 = ['Admin2', 'Province_State', 'Country_Region', 'Lat', 'Long_', 'Combined_Key', 'Date', 'Confirmed', 'Deaths']
usecols9 = ['Date', 'Temperature (° F) - Avg', 'Dew Point (° F) - Avg', 'Humidity (%) - Avg', 'Wind Speed (mph) - Avg', 
           'Pressure (Hg) - Avg', 'Precipitation (in) Total']
    
#load data
cs = pd.read_csv("CUSTOMERS.csv", usecols=usecols1)
csc = pd.read_csv("CUSTOMER_CATEGORY.csv", usecols=usecols2)
ps = pd.read_csv("PERSONS.csv", usecols=usecols3)
tr = pd.read_csv("TRANSACTIONS.csv", usecols=usecols4)
sp = pd.read_csv("SUPPLIERS.csv", usecols=usecols5)
spc = pd.read_csv("SUPPLIER_CATEGORY.csv", usecols=usecols6)
spt = pd.read_csv("SUPPLIER_TRANSACTION.csv", usecols=usecols7)
cov = pd.read_csv("Usa_county_covid 19.csv", usecols=usecols8)
wt = pd.read_csv("Weather.csv", encoding='latin1', usecols=usecols9)  #I specified the encoding because "utf-8" could not decode its special characters


In [3]:
#I will rename some columns to clean up the data a little bit
cs.rename(columns={"PrimaryContactPersonID":"PersonID"}, inplace=True)
csc.rename(columns={"CustomerCategoryName":"Goods_Category"}, inplace=True)
ps.rename(columns={"FullName":"Agent_Name"}, inplace=True)
tr.rename(columns={"TransactionDate":"Date"}, inplace=True)
sp.rename(columns={"PrimaryContactPersonID":"PersonID"}, inplace=True)
spc.rename(columns={"SupplierCategoryName":"Goods_Category"}, inplace=True)

In [4]:
cs

CustomerID                        CustomerName  CustomerCategoryID  \
0             1         Tailspin Toys (Head Office)                   3   
1             2       Tailspin Toys (Sylvanite, MT)                   3   
2             3  Tailspin Toys (Peeples Valley, AZ)                   3   
3             4  Tailspin Toys (Medicine Lodge, KS)                   3   
4             5         Tailspin Toys (Gasport, NY)                   3   
..          ...                                 ...                 ...   
658        1057                     Ganesh Majumdar                   5   
659        1058                      Jaroslav Fisar                   6   
660        1059                     Jibek Juniskyzy                   4   
661        1060                     Anand Mudaliyar                   7   
662        1061                        Agrita Abele                   5   

     BuyingGroupID  PersonID  
0              1.0      1001  
1              1.0      1003  
2              1.0      1005  
3              1.0      1007  
4              1.0      1009  
..             ...       ...  
658            NaN      3257  
659            NaN      3258  
660            NaN      3259  
661            NaN      3260  
662            NaN      3261  

[663 rows x 5 columns]

In [5]:
csc

CustomerCategoryID    Goods_Category
0                   1             Agent
1                   2        Wholesaler
2                   3      Novelty Shop
3                   4       Supermarket
4                   5    Computer Store
5                   6        Gift Store
6                   7         Corporate
7                   8  General Retailer

In [6]:
ps

PersonID            Agent_Name
0            1  Data Conversion Only
1            2        Kayla Woodcock
2            3         Hudson Onslow
3            4         Isabella Rupp
4            5           Eva Muirden
...        ...                   ...
1106      3257       Ganesh Majumdar
1107      3258        Jaroslav Fisar
1108      3259       Jibek Juniskyzy
1109      3260       Anand Mudaliyar
1110      3261          Agrita Abele

[1111 rows x 2 columns]

In [7]:
tr

CustomerID        Date  TransactionAmount  IsFinalized
0             832  2016-11-01           952200.0            1
1             803  2016-11-01           167670.0            1
2               1  2016-11-01            37260.0            1
3               1  2016-11-01           184312.8            1
4             905  2016-11-01           291456.0            1
...           ...         ...                ...          ...
97142         401  2020-03-31           178806.6            0
97143         911  2020-03-31           683203.5            0
97144         401  2020-03-31           575253.0            0
97145           1  2020-03-31           366017.4            0
97146         401  2020-03-31           699287.4            0

[97147 rows x 4 columns]

In [8]:
sp

SupplierID              SupplierName  SupplierCategoryID  PersonID
0            1       A Datum Corporation                   2        21
1            2             Contoso, Ltd.                   2        23
2            3    Consolidated Messenger                   6        25
3            4            Fabrikam, Inc.                   4        27
4            5  Graphic Design Institute                   2        29
5            6       Humongous Insurance                   9        31
6            7             Litware, Inc.                   5        33
7            8        Lucerne Publishing                   2        35
8            9            Nod Publishers                   2        37
9           10   Northwind Electric Cars                   3        39
10          11             Trey Research                   8        41
11          12         The Phone Company                   2        43
12          13            Woodgrove Bank                   7        45

In [9]:
spc

SupplierCategoryID               Goods_Category
0                   1             Other Wholesaler
1                   2       Novelty Goods Supplier
2                   3                 Toy Supplier
3                   4            Clothing Supplier
4                   5           Packaging Supplier
5                   6    Courier Services Supplier
6                   7  Financial Services Supplier
7                   8  Marketing Services Supplier
8                   9  Insurance Services Supplier

In [10]:
spt

SupplierID  PurchaseOrderID  TransactionAmount  IsFinalized
0              2              1.0       1.297908e+05            1
1              4              2.0       8.997048e+06            1
2              5              3.0       1.134580e+06            1
3              7              4.0       1.758722e+07            1
4             10              5.0       1.451795e+07            1
...          ...              ...                ...          ...
2433           4           2070.0       2.168532e+05            1
2434           4              NaN       1.508708e+09            1
2435           7              NaN       6.840143e+08            1
2436           4           2071.0       3.032468e+08            0
2437           7           2072.0       1.376650e+08            0

[2438 rows x 4 columns]

In [11]:
cov

Admin2 Province_State Country_Region        Lat      Long_  \
0          Albany       New York             US  42.600603 -73.977239   
1        Allegany       New York             US  42.257484 -78.027505   
2           Bronx       New York             US  40.852093 -73.862828   
3          Broome       New York             US  42.159032 -75.813261   
4     Cattaraugus       New York             US  42.247782 -78.679231   
...           ...            ...            ...        ...        ...   
4667  Westchester       New York             US  41.162784 -73.757417   
4668      Wyoming       New York             US  42.701451 -78.221996   
4669        Yates       New York             US  42.635055 -77.103699   
4670    Out of NY       New York             US   0.000000   0.000000   
4671   Unassigned       New York             US   0.000000   0.000000   

                   Combined_Key        Date  Confirmed  Deaths  
0          Albany, New York, US     1/22/20          0       0  
1        Allegany, New York, US     1/22/20          0       0  
2           Bronx, New York, US     1/22/20          0       0  
3          Broome, New York, US     1/22/20          0       0  
4     Cattaraugus, New York, US     1/22/20          0       0  
...                         ...         ...        ...     ...  
4667  Westchester, New York, US  04/03/2020      12351      67  
4668      Wyoming, New York, US  04/03/2020         15       1  
4669        Yates, New York, US  04/03/2020          1       0  
4670    Out of NY, New York, US  04/03/2020          0       0  
4671   Unassigned, New York, US  04/03/2020          0     955  

[4672 rows x 9 columns]

In [12]:
wt

Date  Temperature (° F) - Avg  Dew Point (° F) - Avg  \
0     01/01/2016                     40.7                   24.0   
1     02/01/2016                     37.3                   18.5   
2     03/01/2016                     41.0                   23.2   
3     04/01/2016                     29.3                   10.3   
4     05/01/2016                     20.7                   -2.2   
...          ...                      ...                    ...   
1558  07/04/2020                     54.7                   34.8   
1559  08/04/2020                     54.2                   49.5   
1560  09/04/2020                     50.2                   43.6   
1561  10/04/2020                     43.9                   28.2   
1562  11/04/2020                     43.8                   21.4   

      Humidity (%) - Avg  Wind Speed (mph) - Avg  Pressure (Hg) - Avg  \
0                   51.7                    14.4                 30.0   
1                   46.8                    13.3                 30.0   
2                   50.3                    14.3                 29.9   
3                   45.7                    19.3                 30.0   
4                   38.5                    13.2                 30.6   
...                  ...                     ...                  ...   
1558                50.8                    10.4                 29.9   
1559                86.0                     9.6                 29.6   
1560                81.8                    13.7                 29.4   
1561                55.0                    24.4                 29.5   
1562                41.7                    18.3                 29.9   

      Precipitation (in) Total  
0                         0.00  
1                         0.00  
2                         0.00  
3                         0.00  
4                         0.00  
...                        ...  
1558                      0.00  
1559                      0.18  
1560                      0.00  
1561                      0.13  
1562                      0.00  

[1563 rows x 7 columns]

### I want to merge all the sales tables together, to make it easier for analysis

In [13]:
sales1=cs.merge(csc,on='CustomerCategoryID', how="left") #I merged on the left because I would want to retain all the data in the first dataframe and also use the information on the second table
sales2=ps.merge(sales1,on='PersonID', how="left")
sales=tr.merge(sales2,on='CustomerID', how="left")
sales

CustomerID        Date  TransactionAmount  IsFinalized  PersonID  \
0             832  2016-11-01           952200.0            1      3032   
1             803  2016-11-01           167670.0            1      3003   
2               1  2016-11-01            37260.0            1      1001   
3               1  2016-11-01           184312.8            1      1001   
4             905  2016-11-01           291456.0            1      3105   
...           ...         ...                ...          ...       ...   
97142         401  2020-03-31           178806.6            0      2001   
97143         911  2020-03-31           683203.5            0      3111   
97144         401  2020-03-31           575253.0            0      2001   
97145           1  2020-03-31           366017.4            0      1001   
97146         401  2020-03-31           699287.4            0      2001   

             Agent_Name                 CustomerName  CustomerCategoryID  \
0       Aakriti Byrraju              Aakriti Byrraju                 4.0   
1            Bala Dixit                   Bala Dixit                 3.0   
2        Waldemar Fisar  Tailspin Toys (Head Office)                 3.0   
3        Waldemar Fisar  Tailspin Toys (Head Office)                 3.0   
4          Sara Huiting                 Sara Huiting                 4.0   
...                 ...                          ...                 ...   
97142  Olga Alexandrova   Wingtip Toys (Head Office)                 3.0   
97143       Liidia Lepp                  Liidia Lepp                 6.0   
97144  Olga Alexandrova   Wingtip Toys (Head Office)                 3.0   
97145    Waldemar Fisar  Tailspin Toys (Head Office)                 3.0   
97146  Olga Alexandrova   Wingtip Toys (Head Office)                 3.0   

       BuyingGroupID Goods_Category  
0                NaN    Supermarket  
1                NaN   Novelty Shop  
2                1.0   Novelty Shop  
3                1.0   Novelty Shop  
4                NaN    Supermarket  
...              ...            ...  
97142            2.0   Novelty Shop  
97143            NaN     Gift Store  
97144            2.0   Novelty Shop  
97145            1.0   Novelty Shop  
97146            2.0   Novelty Shop  

[97147 rows x 10 columns]

### I also want to do the same for the suppliers tables

In [14]:
suppliers1=sp.merge(spc,on='SupplierCategoryID', how="left") #I merged on the left because I would want to retain all the data in the first dataframe and also use the information on the second table
suppliers=spt.merge(suppliers1,on='SupplierID', how="left")
suppliers

SupplierID  PurchaseOrderID  TransactionAmount  IsFinalized  \
0              2              1.0       1.297908e+05            1   
1              4              2.0       8.997048e+06            1   
2              5              3.0       1.134580e+06            1   
3              7              4.0       1.758722e+07            1   
4             10              5.0       1.451795e+07            1   
...          ...              ...                ...          ...   
2433           4           2070.0       2.168532e+05            1   
2434           4              NaN       1.508708e+09            1   
2435           7              NaN       6.840143e+08            1   
2436           4           2071.0       3.032468e+08            0   
2437           7           2072.0       1.376650e+08            0   

                  SupplierName  SupplierCategoryID  PersonID  \
0                Contoso, Ltd.                   2        23   
1               Fabrikam, Inc.                   4        27   
2     Graphic Design Institute                   2        29   
3                Litware, Inc.                   5        33   
4      Northwind Electric Cars                   3        39   
...                        ...                 ...       ...   
2433            Fabrikam, Inc.                   4        27   
2434            Fabrikam, Inc.                   4        27   
2435             Litware, Inc.                   5        33   
2436            Fabrikam, Inc.                   4        27   
2437             Litware, Inc.                   5        33   

              Goods_Category  
0     Novelty Goods Supplier  
1          Clothing Supplier  
2     Novelty Goods Supplier  
3         Packaging Supplier  
4               Toy Supplier  
...                      ...  
2433       Clothing Supplier  
2434       Clothing Supplier  
2435      Packaging Supplier  
2436       Clothing Supplier  
2437      Packaging Supplier  

[2438 rows x 8 columns]

## Data Inspection and Wrangling

### Sales

In [15]:
#check missing values
sales.isnull().sum()

CustomerID                0
Date                      0
TransactionAmount         0
IsFinalized               0
PersonID                  0
Agent_Name                0
CustomerName              0
CustomerCategoryID        0
BuyingGroupID         50530
Goods_Category            0
dtype: int64

#### We have 50530 missing values in BuyingGroupID, but from the excel file that explained the variables, the missing values is indicating that the customer(s) is an individual and not in a group. The analysis we are going to do will be based on the individual customers, so we have to remove the data of the transactions done in group(s).
#### Also, IsFinalized '0' values has to be dropped too because it indicates the transactions that have not been finalized.

In [16]:
sales['BuyingGroupID'].unique()

array([nan,  1.,  2.])

In [17]:
sales.replace({'BuyingGroupID': {np.NaN: 10}}, inplace = True) #I replaced it with 10 (you can use any number aside 1 or 2) to change it to notnull value
sales.replace({'BuyingGroupID': {1.0: np.NaN}}, inplace = True) #I changed it to null value so that I can easily drop it
sales.replace({'BuyingGroupID': {2.0: np.NaN}}, inplace = True)
sales.replace({'IsFinalized': {0: np.NaN}}, inplace = True)
sales

CustomerID        Date  TransactionAmount  IsFinalized  PersonID  \
0             832  2016-11-01           952200.0          1.0      3032   
1             803  2016-11-01           167670.0          1.0      3003   
2               1  2016-11-01            37260.0          1.0      1001   
3               1  2016-11-01           184312.8          1.0      1001   
4             905  2016-11-01           291456.0          1.0      3105   
...           ...         ...                ...          ...       ...   
97142         401  2020-03-31           178806.6          NaN      2001   
97143         911  2020-03-31           683203.5          NaN      3111   
97144         401  2020-03-31           575253.0          NaN      2001   
97145           1  2020-03-31           366017.4          NaN      1001   
97146         401  2020-03-31           699287.4          NaN      2001   

             Agent_Name                 CustomerName  CustomerCategoryID  \
0       Aakriti Byrraju              Aakriti Byrraju                 4.0   
1            Bala Dixit                   Bala Dixit                 3.0   
2        Waldemar Fisar  Tailspin Toys (Head Office)                 3.0   
3        Waldemar Fisar  Tailspin Toys (Head Office)                 3.0   
4          Sara Huiting                 Sara Huiting                 4.0   
...                 ...                          ...                 ...   
97142  Olga Alexandrova   Wingtip Toys (Head Office)                 3.0   
97143       Liidia Lepp                  Liidia Lepp                 6.0   
97144  Olga Alexandrova   Wingtip Toys (Head Office)                 3.0   
97145    Waldemar Fisar  Tailspin Toys (Head Office)                 3.0   
97146  Olga Alexandrova   Wingtip Toys (Head Office)                 3.0   

       BuyingGroupID Goods_Category  
0               10.0    Supermarket  
1               10.0   Novelty Shop  
2                NaN   Novelty Shop  
3                NaN   Novelty Shop  
4               10.0    Supermarket  
...              ...            ...  
97142            NaN   Novelty Shop  
97143           10.0     Gift Store  
97144            NaN   Novelty Shop  
97145            NaN   Novelty Shop  
97146            NaN   Novelty Shop  

[97147 rows x 10 columns]

In [18]:
sales.isnull().sum()

CustomerID                0
Date                      0
TransactionAmount         0
IsFinalized              84
PersonID                  0
Agent_Name                0
CustomerName              0
CustomerCategoryID        0
BuyingGroupID         46617
Goods_Category            0
dtype: int64

#### Now, we can go ahead and drop all the rows of the missing values

In [19]:
sales = sales.dropna() 
sales

CustomerID        Date  TransactionAmount  IsFinalized  PersonID  \
0             832  2016-11-01          952200.00          1.0      3032   
1             803  2016-11-01          167670.00          1.0      3003   
4             905  2016-11-01          291456.00          1.0      3105   
5             976  2016-11-01          178020.00          1.0      3176   
7             964  2016-11-01          106812.00          1.0      3164   
...           ...         ...                ...          ...       ...   
97058        1028  2020-03-31           19561.50          1.0      3228   
97059        1029  2020-03-31           64335.60          1.0      3229   
97060        1040  2020-03-31          231920.64          1.0      3240   
97061        1054  2020-03-31           75247.20          1.0      3254   
97062        1059  2020-03-31          167359.50          1.0      3259   

              Agent_Name       CustomerName  CustomerCategoryID  \
0        Aakriti Byrraju    Aakriti Byrraju                 4.0   
1             Bala Dixit         Bala Dixit                 3.0   
4           Sara Huiting       Sara Huiting                 4.0   
5           Alinne Matos       Alinne Matos                 5.0   
7        Ingrida Zeltina    Ingrida Zeltina                 4.0   
...                  ...                ...                 ...   
97058      Emma Van Zant      Emma Van Zant                 6.0   
97059  Veronika Necesana  Veronika Necesana                 3.0   
97060     Damodar Shenoy     Damodar Shenoy                 3.0   
97061         Emma Salpa         Emma Salpa                 4.0   
97062    Jibek Juniskyzy    Jibek Juniskyzy                 4.0   

       BuyingGroupID  Goods_Category  
0               10.0     Supermarket  
1               10.0    Novelty Shop  
4               10.0     Supermarket  
5               10.0  Computer Store  
7               10.0     Supermarket  
...              ...             ...  
97058           10.0      Gift Store  
97059           10.0    Novelty Shop  
97060           10.0    Novelty Shop  
97061           10.0     Supermarket  
97062           10.0     Supermarket  

[50489 rows x 10 columns]

In [20]:
#checking data types
sales.dtypes

CustomerID              int64
Date                   object
TransactionAmount     float64
IsFinalized           float64
PersonID                int64
Agent_Name             object
CustomerName           object
CustomerCategoryID    float64
BuyingGroupID         float64
Goods_Category         object
dtype: object

In [21]:
#They are okay except 'Date', let's change it to date type
sales['Date'] = pd.to_datetime(sales['Date'])
sales.dtypes

CustomerID                     int64
Date                  datetime64[ns]
TransactionAmount            float64
IsFinalized                  float64
PersonID                       int64
Agent_Name                    object
CustomerName                  object
CustomerCategoryID           float64
BuyingGroupID                float64
Goods_Category                object
dtype: object

In [22]:
#let us split the 'TransactionDate' to have month, quarter and year column s.loc[[1, 2]]
sales['Year']= sales['Date'].dt.year
sales['Quarter']= sales['Date'].dt.quarter
sales['Month']= sales['Date'].dt.month
sales


CustomerID       Date  TransactionAmount  IsFinalized  PersonID  \
0             832 2016-11-01          952200.00          1.0      3032   
1             803 2016-11-01          167670.00          1.0      3003   
4             905 2016-11-01          291456.00          1.0      3105   
5             976 2016-11-01          178020.00          1.0      3176   
7             964 2016-11-01          106812.00          1.0      3164   
...           ...        ...                ...          ...       ...   
97058        1028 2020-03-31           19561.50          1.0      3228   
97059        1029 2020-03-31           64335.60          1.0      3229   
97060        1040 2020-03-31          231920.64          1.0      3240   
97061        1054 2020-03-31           75247.20          1.0      3254   
97062        1059 2020-03-31          167359.50          1.0      3259   

              Agent_Name       CustomerName  CustomerCategoryID  \
0        Aakriti Byrraju    Aakriti Byrraju                 4.0   
1             Bala Dixit         Bala Dixit                 3.0   
4           Sara Huiting       Sara Huiting                 4.0   
5           Alinne Matos       Alinne Matos                 5.0   
7        Ingrida Zeltina    Ingrida Zeltina                 4.0   
...                  ...                ...                 ...   
97058      Emma Van Zant      Emma Van Zant                 6.0   
97059  Veronika Necesana  Veronika Necesana                 3.0   
97060     Damodar Shenoy     Damodar Shenoy                 3.0   
97061         Emma Salpa         Emma Salpa                 4.0   
97062    Jibek Juniskyzy    Jibek Juniskyzy                 4.0   

       BuyingGroupID  Goods_Category  Year  Quarter  Month  
0               10.0     Supermarket  2016        4     11  
1               10.0    Novelty Shop  2016        4     11  
4               10.0     Supermarket  2016        4     11  
5               10.0  Computer Store  2016        4     11  
7               10.0     Supermarket  2016        4     11  
...              ...             ...   ...      ...    ...  
97058           10.0      Gift Store  2020        1      3  
97059           10.0    Novelty Shop  2020        1      3  
97060           10.0    Novelty Shop  2020        1      3  
97061           10.0     Supermarket  2020        1      3  
97062           10.0     Supermarket  2020        1      3  

[50489 rows x 13 columns]

In [23]:
#Now we can export the sales data to be explored in a seperate notebook
sales.to_csv('Sales.csv',index=False)

### Suppliers

In [24]:
#check missing values
suppliers.isnull().sum()

SupplierID              0
PurchaseOrderID       366
TransactionAmount       0
IsFinalized             0
SupplierName            0
SupplierCategoryID      0
PersonID                0
Goods_Category          0
dtype: int64

#### We have to remove the rows that do not have PurchaseOrderID because the transaction is invalid without an order ID.  Just like we did in sales,  IsFinalized '0' values has to be dropped too because it indicates the transactions that have not been finalized.

In [25]:
suppliers.replace({'IsFinalized': {0: np.NaN}}, inplace = True)
suppliers = suppliers.dropna()
suppliers

SupplierID  PurchaseOrderID  TransactionAmount  IsFinalized  \
0              2              1.0       1.297908e+05          1.0   
1              4              2.0       8.997048e+06          1.0   
2              5              3.0       1.134580e+06          1.0   
3              7              4.0       1.758722e+07          1.0   
4             10              5.0       1.451795e+07          1.0   
...          ...              ...                ...          ...   
2429           4           2066.0       3.017278e+08          1.0   
2430           7           2067.0       1.371867e+08          1.0   
2431           4           2068.0       3.026701e+08          1.0   
2432           7           2069.0       1.373327e+08          1.0   
2433           4           2070.0       2.168532e+05          1.0   

                  SupplierName  SupplierCategoryID  PersonID  \
0                Contoso, Ltd.                   2        23   
1               Fabrikam, Inc.                   4        27   
2     Graphic Design Institute                   2        29   
3                Litware, Inc.                   5        33   
4      Northwind Electric Cars                   3        39   
...                        ...                 ...       ...   
2429            Fabrikam, Inc.                   4        27   
2430             Litware, Inc.                   5        33   
2431            Fabrikam, Inc.                   4        27   
2432             Litware, Inc.                   5        33   
2433            Fabrikam, Inc.                   4        27   

              Goods_Category  
0     Novelty Goods Supplier  
1          Clothing Supplier  
2     Novelty Goods Supplier  
3         Packaging Supplier  
4               Toy Supplier  
...                      ...  
2429       Clothing Supplier  
2430      Packaging Supplier  
2431       Clothing Supplier  
2432      Packaging Supplier  
2433       Clothing Supplier  

[2070 rows x 8 columns]

In [26]:
suppliers.dtypes

SupplierID              int64
PurchaseOrderID       float64
TransactionAmount     float64
IsFinalized           float64
SupplierName           object
SupplierCategoryID      int64
PersonID                int64
Goods_Category         object
dtype: object

In [27]:
suppliers.to_csv('DSuppliers.csv',index=False)

### Covid

In [28]:
cov.isnull().sum()

Admin2            0
Province_State    0
Country_Region    0
Lat               0
Long_             0
Combined_Key      0
Date              0
Confirmed         0
Deaths            0
dtype: int64

In [29]:
cov.dtypes

Admin2             object
Province_State     object
Country_Region     object
Lat               float64
Long_             float64
Combined_Key       object
Date               object
Confirmed           int64
Deaths              int64
dtype: object

In [30]:
cov['Date']= pd.to_datetime(cov['Date'])
cov.dtypes

Admin2                    object
Province_State            object
Country_Region            object
Lat                      float64
Long_                    float64
Combined_Key              object
Date              datetime64[ns]
Confirmed                  int64
Deaths                     int64
dtype: object

In [31]:
cov['TransactionYear']= cov['Date'].dt.year
cov['TransactionQtr']= cov['Date'].dt.quarter
cov['TransactionMth']= cov['Date'].dt.month
cov

Admin2 Province_State Country_Region        Lat      Long_  \
0          Albany       New York             US  42.600603 -73.977239   
1        Allegany       New York             US  42.257484 -78.027505   
2           Bronx       New York             US  40.852093 -73.862828   
3          Broome       New York             US  42.159032 -75.813261   
4     Cattaraugus       New York             US  42.247782 -78.679231   
...           ...            ...            ...        ...        ...   
4667  Westchester       New York             US  41.162784 -73.757417   
4668      Wyoming       New York             US  42.701451 -78.221996   
4669        Yates       New York             US  42.635055 -77.103699   
4670    Out of NY       New York             US   0.000000   0.000000   
4671   Unassigned       New York             US   0.000000   0.000000   

                   Combined_Key       Date  Confirmed  Deaths  \
0          Albany, New York, US 2020-01-22          0       0   
1        Allegany, New York, US 2020-01-22          0       0   
2           Bronx, New York, US 2020-01-22          0       0   
3          Broome, New York, US 2020-01-22          0       0   
4     Cattaraugus, New York, US 2020-01-22          0       0   
...                         ...        ...        ...     ...   
4667  Westchester, New York, US 2020-04-03      12351      67   
4668      Wyoming, New York, US 2020-04-03         15       1   
4669        Yates, New York, US 2020-04-03          1       0   
4670    Out of NY, New York, US 2020-04-03          0       0   
4671   Unassigned, New York, US 2020-04-03          0     955   

      TransactionYear  TransactionQtr  TransactionMth  
0                2020               1               1  
1                2020               1               1  
2                2020               1               1  
3                2020               1               1  
4                2020               1               1  
...               ...             ...             ...  
4667             2020               2               4  
4668             2020               2               4  
4669             2020               2               4  
4670             2020               2               4  
4671             2020               2               4  

[4672 rows x 12 columns]

In [32]:
cov.to_csv('Covid.csv',index=False)

### Weather

In [33]:
wt.dtypes

Date                         object
Temperature (° F) - Avg     float64
Dew Point (° F) - Avg       float64
Humidity (%) - Avg          float64
Wind Speed (mph) - Avg      float64
Pressure (Hg) - Avg         float64
Precipitation (in) Total    float64
dtype: object

In [34]:
wt['Date']= pd.to_datetime(wt['Date'])
wt.dtypes

Date                        datetime64[ns]
Temperature (° F) - Avg            float64
Dew Point (° F) - Avg              float64
Humidity (%) - Avg                 float64
Wind Speed (mph) - Avg             float64
Pressure (Hg) - Avg                float64
Precipitation (in) Total           float64
dtype: object

In [35]:
#let us split the 'Date' to have day, month, quarter and year column
wt['Year']= wt['Date'].dt.year
wt['Quarter']= wt['Date'].dt.quarter
wt['Month']= wt['Date'].dt.month
wt['Day']= wt['Date'].dt.day
wt

Date  Temperature (° F) - Avg  Dew Point (° F) - Avg  \
0    2016-01-01                     40.7                   24.0   
1    2016-02-01                     37.3                   18.5   
2    2016-03-01                     41.0                   23.2   
3    2016-04-01                     29.3                   10.3   
4    2016-05-01                     20.7                   -2.2   
...         ...                      ...                    ...   
1558 2020-07-04                     54.7                   34.8   
1559 2020-08-04                     54.2                   49.5   
1560 2020-09-04                     50.2                   43.6   
1561 2020-10-04                     43.9                   28.2   
1562 2020-11-04                     43.8                   21.4   

      Humidity (%) - Avg  Wind Speed (mph) - Avg  Pressure (Hg) - Avg  \
0                   51.7                    14.4                 30.0   
1                   46.8                    13.3                 30.0   
2                   50.3                    14.3                 29.9   
3                   45.7                    19.3                 30.0   
4                   38.5                    13.2                 30.6   
...                  ...                     ...                  ...   
1558                50.8                    10.4                 29.9   
1559                86.0                     9.6                 29.6   
1560                81.8                    13.7                 29.4   
1561                55.0                    24.4                 29.5   
1562                41.7                    18.3                 29.9   

      Precipitation (in) Total  Year  Quarter  Month  Day  
0                         0.00  2016        1      1    1  
1                         0.00  2016        1      2    1  
2                         0.00  2016        1      3    1  
3                         0.00  2016        2      4    1  
4                         0.00  2016        2      5    1  
...                        ...   ...      ...    ...  ...  
1558                      0.00  2020        3      7    4  
1559                      0.18  2020        3      8    4  
1560                      0.00  2020        3      9    4  
1561                      0.13  2020        4     10    4  
1562                      0.00  2020        4     11    4  

[1563 rows x 11 columns]

In [36]:
wt.to_csv('DWeather.csv',index=False)